wave_viewer.m

```diff
@@ -1940,6 +1946,61 @@ function expand_btw_ax_tmarkers(ax,tAx,fAx)
 end
 end
 
+function expand_btw_ax_uev(ax,tAx,fAx) 
+[~,t_spec,~,t_user_events] = get_ax_tmarker_times(ax);
+fig_params = get_fig_params;
+if length(t_user_events) >= 2
+    tmarker_buffer = 0.05;
+    t_low = t_user_events(1) - tmarker_buffer;
+    t_hi = t_user_events(end) + tmarker_buffer;
+elseif length(t_user_events) == 1
+    tmarker_buffer = 0.2;
+    t_low = t_user_events(1) - tmarker_buffer;
+    t_hi = t_user_events(1) + tmarker_buffer;
+else
+    return;
+end
+
+%finds proper axis boundaries and tmarker times
+tAx_latest_start_time = intmin;
+tAx_earliest_end_time = intmax;
+for i = 1:length(tAx) %find the axis with the most restrictive start and end times
+    if tAx(i).UserData.taxis(1) > tAx_latest_start_time
+        tAx_latest_start_time = tAx(i).UserData.taxis(1);
+    end
+    if tAx(i).UserData.taxis(end) < tAx_earliest_end_time
+        tAx_earliest_end_time = tAx(i).UserData.taxis(end);
+    end
+end
+t_low_ax_min = tAx_latest_start_time + fig_params.tmarker_init_border*tmarker_buffer*2; %set floor and ceiling so all axes are aligned
+t_hi_ax_max = tAx_earliest_end_time - fig_params.tmarker_init_border*tmarker_buffer*2;
+if t_low <= t_low_ax_min %conform to floor and ceiling if necessary
+    t_low = t_low_ax_min;
+end
+if t_hi >= t_hi_ax_max
+    t_hi = t_hi_ax_max;
+end
```

Comment:
1965-1974 finds most restrictive axis.
1975-1976 adds buffer for the bit outside the time markers.